src/prng.rs

```diff
@@ -76,23 +74,51 @@ where
             // Seek to the next chunk of the buffer that encodes an element of F.
             for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
                 let j = i + F::ENCODED_SIZE;
+
+                if j > self.buffer.len() {
+                    break;
+                }
+
+                self.buffer_index = j;
+
                 if let Some(x) = match F::try_from_random(&self.buffer[i..j]) {
                     Ok(x) => Some(x),
                     Err(FieldError::ModulusOverflow) => None, // reject this sample
                     Err(err) => panic!("unexpected error: {err}"),
                 } {
-                    // Set the buffer index to the next chunk.
-                    self.buffer_index = j;
-                    self.output_written += 1;
                     return x;
                 }
             }
 
-            // Refresh buffer with the next chunk of PRG output.
-            self.seed_stream.fill(&mut self.buffer);
+            // Refresh buffer with the next chunk of PRG output, filling the front of the buffer
+            // with the leftovers. This ensures continuity of the seeds stream after converting the
+            // `Prng` to a new field type via `into_new_field()`.
+            let left_over = self.buffer.len() - self.buffer_index;
+            for i in 0..left_over {
+                self.buffer[i] = self.buffer[self.buffer_index + i];
+            }
```

Comment:
nit: This could use the `copy_within()` method, which would probably hoist some bounds checks

Reply:
Wow, that was the best thing that happened to me all day. What a find!